azurerm/internal/services/loadbalancer/backend_address_pool_address_resource.go

```diff
@@ -0,0 +1,335 @@
+				return fmt.Errorf("retrieving %s: `properties` was nil", *poolId)
+			}
+
+			addresses := make([]network.LoadBalancerBackendAddress, 0)
+			if pool.BackendAddressPoolPropertiesFormat.LoadBalancerBackendAddresses != nil {
+				addresses = *pool.BackendAddressPoolPropertiesFormat.LoadBalancerBackendAddresses
+			}
+
+			metadata.Logger.Infof("checking for existing %s..", id)
+			for _, address := range addresses {
+				if address.Name == nil {
+					continue
+				}
+
+				if *address.Name == id.AddressName {
+					return metadata.ResourceRequiresImport(r.ResourceType(), id)
+				}
+			}
+
+			addresses = append(addresses, network.LoadBalancerBackendAddress{
+				LoadBalancerBackendAddressPropertiesFormat: &network.LoadBalancerBackendAddressPropertiesFormat{
+					IPAddress: utils.String(model.IPAddress),
+					VirtualNetwork: &network.SubResource{
+						ID: utils.String(model.VirtualNetworkId),
+					},
+				},
+				Name: utils.String(id.AddressName),
+			})
+			pool.BackendAddressPoolPropertiesFormat.LoadBalancerBackendAddresses = &addresses
+
+			metadata.Logger.Infof("adding %s..", id)
+			future, err := client.CreateOrUpdate(ctx, id.ResourceGroup, id.LoadBalancerName, id.BackendAddressPoolName, pool)
+			if err != nil {
+				return fmt.Errorf("updating %s: %+v", id, err)
+			}
+			metadata.Logger.Infof("waiting for update %s..", id)
+			if err := future.WaitForCompletionRef(ctx, client.Client); err != nil {
+				return fmt.Errorf("waiting for update of %s: %+v", id, err)
+			}
+			metadata.SetID(id)
+			return nil
+		},
+		Timeout: 30 * time.Minute,
+	}
+}
+
+func (r BackendAddressPoolAddressResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.LoadBalancers.LoadBalancerBackendAddressPoolsClient
+			id, err := parse.BackendAddressPoolAddressID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			pool, err := client.Get(ctx, id.ResourceGroup, id.LoadBalancerName, id.BackendAddressPoolName)
+			if err != nil {
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+			if pool.BackendAddressPoolPropertiesFormat == nil {
+				return fmt.Errorf("retrieving %s: `properties` was nil", *id)
+			}
+
+			var backendAddress *network.LoadBalancerBackendAddress
+			if pool.BackendAddressPoolPropertiesFormat.LoadBalancerBackendAddresses != nil {
+				for _, address := range *pool.BackendAddressPoolPropertiesFormat.LoadBalancerBackendAddresses {
+					if address.Name == nil {
+						continue
+					}
+
+					if *address.Name == id.AddressName {
+						backendAddress = &address
+						break
+					}
+				}
+			}
+			if backendAddress == nil {
+				metadata.Logger.Warnf("%s was not found - removing from state!")
+				return metadata.MarkAsGone()
+			}
+
+			backendAddressPoolId := parse.NewLoadBalancerBackendAddressPoolID(id.SubscriptionId, id.ResourceGroup, id.LoadBalancerName, id.BackendAddressPoolName)
```

Comment:
We might need further ensure this backend_address_pool_address resource is the IP based address, rather than the NIC based address. For the NIC based addresses, they only have `name` and users are not able to set the other arguments.

The check might be appropriated to do in the `CustomImporter()`?